doc/proposals/scorecard-custom-tests-2.md

```diff
@@ -0,0 +1,261 @@
+
+ * A ConfigMap containing the required test configuration for the test.
+ * A PodSpec with the containers being the bundle image, the test image, and an init container that copies the bundle files from the bundle image into the test image at a specific place (using a shared emptyDir volume mount) (e.g. /scorecard/bundle)
+ * We would also mount the ConfigMap into the test image container (e.g. /scorecard/config)
+
+For this, scorecard would need to support bundle images which is included as a User Story in this proposal.  If the bundle image were to contain the scorecard configuration it might be possble to use that for running the scorecard test as opposed to having a scorecard configuration outside of the bundle image being used.
+
+#### Test Output
+
+Custom tests are expected to produce log output that conforms to the
+scorecard v1alpha2 Test Result output.  This allows scorecard to parse
+the custom test log output and aggregate the results like any other
+scorecard test.
+
+The scorecard expected output would be in JSON, and would match the following definition:  https://github.com/operator-framework/operator-sdk/blob/master/pkg/apis/scorecard/v1alpha2/types.go#L36
+
+The scorecard v1alpha2 Test Result output contains information such as
+pass or fail status and suggestions/errors produced by the test that
+the end user would want to see in the aggregated scorecard results.
+
+
+### Custom Test Development Model
+
+The developer of a custom test would have a workflow similar to:
+ 
+ * create a Pod manifest to run their custom test image, copying a sample Pod manifest from the scorecard examples in the SDK repository
+ * write a custom test binary in the language of their choice, sample custom test within the SDK repository would be available as a starting point
+ * test the custom test binary locally (out-of-cluster) to make sure the output is produced that matches v1alpha2 ScorecardTestResult format
+ * when the test binary works as expected, the developer would build and push the test image to a container registry accessible to the scorecard test environment
+ * the scorecard configuration would be updated to reference the custom test Pod manifest location
+ * the scorecard would discover the custom test based on the scorecard configuration and also the custom test labels applied by the end-user within the custom test Pod manifest
+
+### Custom Test Packaging and Configuration
+
+In this design proposal, custom tests are Pod manifests in YAML that run a custom test image.
+
+Note that it would also be possible to add a resource limit to the scorecard configuration as shown in the example below.  Those resource limits would be applied to custom tests that are executed allowing for some control as to how much resources are available for the test execution.
+
+Custom tests (i.e. Pod manifests) would be accessible from a local disk location.  Custom test locations would be configured from within the scorecard configuration 
```

Comment:
Would this work in the pipeline case where source code is not available?

Reply:
I was originally going to have the scorecard test pod manifests accessible via URL to avoid CI/pipelines from having to have a local copy or have to clone them, but if they are stored in the bundle itself, then that would negate having to have them locally or pull them down via URL.  So, I think making the decision to store scorecard config + test pod manifests inside the bundle image would solve this problem for the CI/pipeline use case.